doc: add arch linux build example

mruddy opened the pull request:
Just adding some notes about building on Arch Linux. Shows which packages are necessary and gives a little info about how the pre-built packages are built.

mruddy committed:
doc: add arch linux setup and build example
[skip ci]

laanwj commented:
Travis failure is unrelated and intermittent (#7463).

BTW for doc-only changes like this you can put `[skip ci]` in the pull request description (not in the title)

MarcoFalke commented:
> put [skip ci] in the pull request description

I think travis no longer fetches this from GitHub. You'd have to put it in the commit body.

MarcoFalke commented:
Not sure if it makes sense to hardcode stuff like "v0.12.0" or "python2" as this will be outdated in one or two months.

mruddy commented:
Cool, I put `[skip ci]` in the commit body and it seems to have worked.
I removed the reference to the tag (tags are around for a long time, but it was not important to what I was documenting so I removed it) and `python2` and reworded things a little.

paveljanik commented on doc/build-unix.md:
```diff
@@ -250,6 +250,24 @@ A list of additional configure flags can be displayed with:
+
+    pacman -S git base-devel boost libevent python
+    git clone https://github.com/bitcoin/bitcoin.git
+    cd bitcoin/
```
save typing, remove / ;-)

mruddy replied:
tab completion :)

paveljanik commented on doc/build-unix.md:
```diff
@@ -250,6 +250,24 @@ A list of additional configure flags can be displayed with:
+This example lists the steps necessary to setup and build a command line only, non-wallet distribution of the latest changes on Arch Linux:
+
+    pacman -S git base-devel boost libevent python
+    git clone https://github.com/bitcoin/bitcoin.git
```
save typing, remove .git at the end

mruddy replied:
copy and paste :)

paveljanik commented on doc/build-unix.md:
```diff
@@ -250,6 +250,24 @@ A list of additional configure flags can be displayed with:
+    git clone https://github.com/bitcoin/bitcoin.git
+    cd bitcoin/
+    ./autogen.sh
+    ./configure --disable-wallet --without-gui --without-miniupnpc
```
Why do you mention miniupnpc here?

mruddy replied:
didn't need it for what i was doing. this was just from my notes. that's all. i also use `make -j 4 check` (it saves a lot of time to run multiple jobs at once) but didn't want to try to describe how to get the number of non-hyperthreaded cores in a generalized way.

paveljanik commented:
ACK https://github.com/bitcoin/bitcoin/pull/7847/commits/fe53a2af6f8dccd53116ae35f28ec3860cc72f19

dcousens commented:
concept ACK

laanwj commented:
> I think travis no longer fetches this from GitHub. You'd have to put it in the commit body.

Please not in the title next time, though.
